app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,22 @@
+class CommentsController < ApplicationController
+  def create
```

Comment:
Kindly create a `new` method that creates a new comment.

Reply:
Hi, @KDlamini could you kindly explain why that's needed? I'm using only the `create` route in this case because I'm having the `comment` form inside the `Posts#show` view which makes more sense to me.

Reply:
> Hi, @KDlamini could you kindly explain why that's needed? I'm using only the `create` route in this case because I'm having the `comment` form inside the `Posts#show` view which makes more sense to me.

Hi @anagudelogu

It's part of rails conventions to have all methods for create, read, update and destroy (CRUD) implemented by the `index`, `new`, `show`, and `destroy` actions in the controller. Although the way you implemented it still works, it's not the best practice. At Microverse, we are teaching you how how to use the best practices therefore you are required to follow the proper rails conventions.

This also means that the form for creating new commnets should also be in its own file, which is the `new.html.erb` file.

Hope that makes sense.

Reply:
Thank you for the explanation @KDlamini but still, as rails offer you a RESTful convention, it doesn't mean that is the best path to follow for every case. 

There are times when it makes sense to avoid using some routes. By convention, `new` exists to present a view to the user so he can submit a form that sends the data to the `create` route. With this in mind, it doesn't really interfere with the flux of the data in the program.

An example would be the `Likes` controller. Guided by what you said, it should have the `new` route too because although it is just a button, it represents a form. But having a separate view only to like the `post` wouldn't provide a great experience for the user, would it?